pyqtorch/hamiltonians/evolution.py

```diff
@@ -142,9 +143,10 @@ class HamiltonianEvolution(Sequence):
     def __init__(
         self,
         generator: TGenerator,
-        time: Tensor | str,
+        time: Tensor | str | ConcretizedCallable,
         qubit_support: Tuple[int, ...] | None = None,
         cache_length: int = 1,
+        duration: float | Tensor = 1.0,
         steps: int = 100,
         solver=SolverType.DP5_SE,
```

Comment:
I remember that I got away without `duration` at some point but then again reverted this decision. I can take a look once more at this.

Reply:
I looked into this issue and it seems that `duration` is unfortunately necessary because `time` has to denote the actual time parameter, thus there has to be an argument to pass the actual duration.